README.md

```diff
@@ -1,2 +1,296 @@
+
+This repository shows you how to use [loopback4-extension-starter](https://github.com/strongloop/loopback4-extension-starter) to write a complex logging extension that requires a [Component](http://loopback.io/doc/en/lb4/Using-components.html), [Decorator](http://loopback.io/doc/en/lb4/Decorators.html), and a [Mixin](http://loopback.io/doc/en/lb4/Mixin.html).
+
+To use the extension, load the component to get access to a `LogFn` that can be used in a sequence to log information. A Mixin allows you to set the application wide logLevel. Only Controller methods configured at or above the logLevel will be logged. 
+
+Possible levels are: DEBUG < INFO < WARN < ERROR < OFF
+
+*Possible levels are represented as numbers but users can use `LOG_LEVEL.${level}` to specify the value instead of using numbers.*
+
+A decorator enables you to provide metadata for Controller methods to set the minimum logLevel.
+
+### Example Usage
+```ts
+import {LogLevelMixin, LogComponent, LOG_LEVEL, log} from 'loopback4-example-log-extension';
+// Other imports ... 
+
+class LogApp extends LogLevelMixin(Application) {
+  constructor() {
+    super({
+      components: [RestComponent, LogComponent],
+      logLevel: LOG_LEVEL.ERROR,
+      controllers: [MyController]
+    });
+  };
+}
+
+class MyController {
+  @log(LOG_LEVEL.WARN)
+  @get('/')
+  hello() {
+    return 'Hello LoopBack';
+  }
+
+  @log(LOG_LEVEL.ERROR)
+  @get('/name')
+  helloName() {
+    return 'Hello Name'
+  }
+}
+```
+
+## Tutorial
+
+Start by cloning [loopback4-extension-starter]() and updating files headers to be `loopback4-example-log-extension` and `package.json`.
+
+Keep all the top level files but modify the `src` directory and `test` directory as follows.
+
+### `/src/index.ts`
+
+Update the exports accordingly here. Typically all files in `/src` are exported. Final version of file should be as follows:
+
+```ts
+export * from './decorators/log.decorator';
+export * from './mixins/log-level.mixin';
+export * from './providers/log-action.provider';
+export * from './providers/log-level.provider';
+export * from './providers/timer.provider';
+export * from './log.component';
+export * from './types';
+export * from './keys';
+```
+
+### `/src/keys.ts`
+
+Define new `Binding` values as well as exporting an `enum` for users to use for setting logLevel. The file will look as follows: 
+
+```ts
+export namespace EXAMPLE_LOG_BINDINGS {
+  export const METADATA = 'example.log.metadata';
+  export const APP_LOG_LEVEL = 'example.log.level';
+  export const TIMER = 'example.log.timer';
+  export const LOG_ACTION = 'example.log.action';
+}
+
+export enum LOG_LEVEL {
+  DEBUG,
+  INFO,
+  WARN,
+  ERROR,
+  OFF,
+}
+```
+
+### `/src/log.component.ts`
+
+Keep this file the same but updating the `Bindings` according to the values we declared in `/src/keys.ts`. This is where providers are exported so they can automatically bind to the appropriate keys. The final version of this file will look as follows:
```

Comment:
I'd also drop `The final version...` since it's implied by the code snippet being up next.